akka-docs-new/src/test/scala/tutorial_3/DeviceGroup.scala

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package tutorial_3
+
+import akka.actor.{ Actor, ActorLogging, ActorRef, Cancellable, Props, Stash, Terminated }
+import tutorial_3.DeviceGroup._
+import tutorial_3.DeviceManager.RequestTrackDevice
+import scala.concurrent.duration._
+
+object DeviceGroup {
+
+  def props(groupId: String): Props = Props(new DeviceGroup(groupId))
+
+  case class RequestDeviceList(requestId: Long)
+  case class ReplyDeviceList(requestId: Long, ids: Set[String])
+
+  case class RequestAllTemperatures(requestId: Long)
+  case class RespondAllTemperatures(requestId: Long, temperatures: Map[String, Option[Double]])
+
+  case class CollectionTimeout(requestId: Long)
+}
+
+class DeviceGroup(groupId: String) extends Actor with ActorLogging with Stash {
+  var deviceIdToActor = Map.empty[String, ActorRef]
+  var actorToDeviceId = Map.empty[ActorRef, String]
+  var nextCollectionId = 0L
+
+  override def preStart(): Unit = log.info(s"DeviceGroup $groupId started")
+
+  override def postStop(): Unit = log.info(s"DeviceGroup $groupId stopped")
+
+  override def receive: Receive = waitingForRequest orElse generalManagement
+
+  def waitingForRequest: Receive = {
+    case RequestAllTemperatures(requestId) =>
+      import context.dispatcher
+
+      val collectionId = nextCollectionId
+      val requester = sender()
+      nextCollectionId += 1
+      val answersSoFar = deviceIdToActor.mapValues(_ => None)
+      context.children.foreach(_ ! Device.ReadTemperature(collectionId))
+      val collectionTimeoutTimer = context.system.scheduler.scheduleOnce(3.seconds, self, CollectionTimeout(collectionId))
+
+      context.become(
+        collectResults(
+          collectionTimeoutTimer,
+          collectionId,
+          requester,
+          requestId,
+          answersSoFar.size,
+          answersSoFar) orElse generalManagement,
+        discardOld = false
+      )
```

Comment:
I forgot the stashing part. But I will create a temporary actor instead as @patriknw suggested which is way superior.